notes-app-server/src/application/setup.ts

```diff
@@ -0,0 +1,28 @@
+import { createRoutes } from "../controllers/routes";
+import { DatabasePort } from "../persist/database.port";
+import { PrismaDBAdapter } from "../persist/prismaDB.adapter";
+import { NotesService } from "../services/notes.service";
+import prisma from "./prisma.singleton";
+
+export type AppDependencies = {
+  notesService: NotesService;
+};
+
+/**
+ * Create the app router
+ */
+export const createAppRouter = () => {
+  const services = initAppDependencies();
+
+  const router = createRoutes(services);
+  return router;
+};
+
+/**
+ * Initialize the app dependencies
+ */
+function initAppDependencies(): AppDependencies {
+  const database: DatabasePort = new PrismaDBAdapter(prisma);
```

Comment:
Here the adapter is used to initialize the specific implementation of the DatabasePort.
Notice how while the new `PrismaDBAdapter` is created, the type here is the `DatabasePort`.